webapp/src/js/services/auth.js

```diff
@@ -148,6 +148,33 @@ angular.module('inboxServices').factory('Auth',
       });
     };
 
+    auth.roles = function(roles) {
+      if (!_.isArray(roles)) {
+        roles = [ roles ];
+      }
+
+      var requiredRoles = getRequired(roles);
+      var disallowedRoles = getDisallowed(roles);
+
+      return getRoles().then(function(userRoles) {
+        if (_.contains(userRoles, '_admin')) {
+          if (disallowedRoles.length > 0) {
+            return authFail('disallowed role(s) found for admin', roles, userRoles);
+          }
+          return $q.resolve();
+        }
+
+        if (!_.every(requiredRoles, _.partial(_.includes, userRoles))) {
+          return authFail('missing required role', roles, userRoles);
+        }
+        if (_.some(disallowedRoles, _.partial(_.includes, userRoles))) {
+          return authFail('found disallowed role', roles, userRoles);
+        }
+
+        return $q.resolve();
+      });
+    };
+
```

Comment:
You've made it generic and reusable which is nice, but I think it'd be a bad idea for anyone to set up role based auth for anything except our special `mm-online` role. To stop devs doing that I think it might be better to make this _less_ generic, for example, `mm-auth-online="true"` and hardcode the role in here somewhere.

Reply:
Yea, I know, but I kind of wanted it to match the other Auth calls.